File: common/src/main/java/com/wdz/common/room/dao/HistoryDao.java
package com.wdz.common.room.dao;

import androidx.room.Dao;
import androidx.room.Query;

import com.wdz.common.base.room.BaseDao;
import com.wdz.common.room.entity.History;

import java.util.List;

import io.reactivex.Flowable;

/**
 * @Author dezhi.wang
 * @Date 2021/2/1 17:13
 */
@Dao
public interface HistoryDao extends BaseDao<History> {

    @Query("SELECT * FROM HISTORY_TAB")
    Flowable<List<History>> loadAllHistory();
}

File: kt_common/src/main/java/com/wdz/ktcommon/room/dao/HistoryDao.kt
package com.wdz.ktcommon.room.dao

import androidx.room.Dao
import androidx.room.Query
import com.wdz.ktcommon.room.base.BaseDao
import com.wdz.ktcommon.room.entity.History
import io.reactivex.Flowable
import kotlinx.coroutines.flow.Flow

/**
 * @Author dezhi.wang
 * @Date 2021/2/1 17:13
 */
@Dao
interface HistoryDao : BaseDao<History> {
    @Query("SELECT * FROM HISTORY_TAB")
    fun loadAllHistory(): Flow<List<History>>
}